### PR TITLE
Fix various issues in python3 formula.

### PR DESCRIPTION
**Pull request by yesimon**
- ```Pathname``` doesn't like concating with nil.
- Fix some other issues with lib_cellar in python3.
- --enable-shared like in python2 formula

### CODE REVIEW

File: Library/Formula/python3.rb [hunk 1]
```diff
@@ -50,10 +50,16 @@ class Python3 < Formula
...
   patch :DATA if build.with? "brewed-tk"
 
+  def lib_cellar
+    if OS.mac?
+      prefix/"Frameworks/Python.framework/Versions/2.7/lib/python2.7"
+    else
+      prefix/"lib/python#{xy}/site-packages"
+    end
+  end
+
```
Comment (sjackman) [inline Library/Formula/python3.rb]: Please use identical code to `python.rb`. It'll make maintenance of the two formulae easier.
See https://github.com/Homebrew/linuxbrew/blob/master/Library/Formula/python.rb#L56-L59

**Comment by sjackman** [comment]
Thanks, Simon.

File: Library/Formula/python3.rb (hunk 1 above)
Comment (yesimon) [inline Library/Formula/python3.rb]: Yeah whoops also have the 2.7 path here.

**Comment by yesimon** [comment]
Updated.

**Comment by sjackman** [comment]
Thanks!

**Pull request closed without merging**